docs/csharp/language-reference/builtin-types/integral-numeric-types.md

```diff
@@ -0,0 +1,117 @@
+|`ulong`|0 to 18,446,744,073,709,551,615|Unsigned 64-bit integer|  
+
+The default value for all integral types is `0`. Each of the integral types has constants named `MinValue` and `MaxValue` for the minimum and maximum value for that type.
+
+Use the <xref:System.Numerics.BigInteger?displayProperty=nameWithType> structure to represent an arbitrarily large signed integer.
+
+## Integral literals
+
+Integral literals can be specified as *decimal literals*, *hexadecimal literals*, or *binary literals*. An example of each is shown below:
+
+```csharp
+var decimalLiteral = 42;
+var hexLiteral = 0x2A;
+var binaryLiteral = 0b_0010_1010;
+```
+
+Decimal literals don't require any prefix. The `x` or `X` prefix signifies a *hexadecimal literal*. The `b` or `B` prefix signifies a *binary literal*. The declaration of `binaryLiteral` demonstrates the use of `_` as a digit separator. Binary literals and the digit separator `_` are supported starting with C# 7.0.
+
+## Type of an integral literal
+
+If an integral literal has no suffix, its type is the first of the following types in which its value can be represented:
+
+1. `int`
+1. `uint`
+1. `long`
+1. `ulong`
+
+You can convert an integral literal to a type with a smaller range than the default using either an assignment or a cast:
+
+```csharp
+byte byteVariable = 42; // type is byte
+var signedByte = (sbyte)42; // type is sbyte.
+```
+
+You can convert an integral literal to a type with a larger range than the default using either assignment, a cast, or a suffix on the literal:
+
+```csharp
+var unsignedLong = 42UL;
+var longVariable = 42L;
+ulong anotherUnsignedLong = 42;
+var anotherLong = (long)42;
+```
+
+The `l` or `L` suffix specifies that the integral literal should be of the `long` type. The `ul` or `UL` suffix specifies the `ulong` type. If the `L` suffix is used on a literal that is greater than 9,223,372,036,854,775,807 (the maximum value of `long`), the value is converted to the `ulong` type. If the value represented by an integer literal exceeds <xref:System.UInt64.MaxValue?displayProperty=nameWithType>, a compiler error [CS1021](../../misc/cs1021.md) occurs. 
+
+> [!NOTE]
+> You can use the lowercase letter "l" as a suffix. However, this generates a compiler warning because the letter "l" is easily confused with the digit "1." Use "L" for clarity.
+
+## Conversions
+
+There's an implicit conversion (called a *widening conversion*) between any two integral types where the destination type can store all values of the source type. For example, there's an implicit conversion from `int` to `long` because the range of `int` values is a proper subset of `long`. There's no implicit conversion from any signed integral type to any unsigned integral type. There are implicit conversions from a smaller unsigned integral type to a larger signed integral type. There's also an implicit conversion from any integral type to any floating-point type.
```

Comment:
```suggestion
There's an implicit conversion (called a *widening conversion*) between any two integral types where the destination type can store all values of the source type. For example, there's an implicit conversion from `int` to `long` because the range of `int` values is a proper subset of `long`. There are implicit conversions from a smaller unsigned integral type to a larger signed integral type. There's also an implicit conversion from any integral type to any floating-point type.  There's no implicit conversion from any signed integral type to any unsigned integral type.
```
I thought that it would be better to list the not-allowed case last rather than in the middle of supported cases.